js/RTCPeerConnection.ts

```diff
@@ -1,878 +1,913 @@
+			this,
+			'addTrack',
+			(RTCPeerConnectionShim as any).prototype_descriptor.addTrack
+		);
+		Object.defineProperty(
+			this,
+			'addStream',
+			(RTCPeerConnectionShim as any).prototype_descriptor.addStream
+		);
+		Object.defineProperty(
+			this,
+			'getLocalStreams',
+			(RTCPeerConnectionShim as any).prototype_descriptor.getLocalStreams
+		);
 
-RTCPeerConnection.prototype = Object.create(EventTarget.prototype);
-RTCPeerConnection.prototype.constructor = RTCPeerConnection;
+		this.pcConfig = fixPcConfig(pcConfig);
 
-Object.defineProperties(RTCPeerConnection.prototype, {
-	localDescription: {
-		// Fix webrtc-adapter TypeError: Attempting to change the getter of an unconfigurable property.
-		configurable: true,
-		get: function () {
-			return this._localDescription;
-		}
-	},
-	connectionState: {
-		get: function () {
-			return this.iceConnectionState;
-		}
-	},
-	onicecandidate: {
-		// Fix webrtc-adapter TypeError: Attempting to change the getter of an unconfigurable property.
-		configurable: true,
-		set: function (callback) {
-			return this.addEventListener('icecandidate', callback);
-		}
-	},
-	onaddstream: {
-		// Fix webrtc-adapter TypeError: Attempting to change the getter of an unconfigurable property.
-		configurable: true,
-		set: function (callback) {
-			return this.addEventListener('addstream', callback);
-		}
-	},
-	ontrack: {
-		// Fix webrtc-adapter TypeError: Attempting to change the getter of an unconfigurable property.
-		configurable: true,
-		set: function (callback) {
-			return this.addEventListener('track', callback);
-		}
-	},
-	oniceconnectionstatechange: {
-		// Fix webrtc-adapter TypeError: Attempting to change the getter of an unconfigurable property.
-		configurable: true,
-		set: function (callback) {
-			return this.addEventListener('iceconnectionstatechange', callback);
-		}
-	},
-	onnegotiationneeded: {
-		// Fix webrtc-adapter TypeError: Attempting to change the getter of an unconfigurable property.
-		configurable: true,
-		set: function (callback) {
-			return this.addEventListener('negotiationneeded', callback);
-		}
+		const onResultOK = (data: PeerConnectionEvent) => this.onEvent(data);
+		exec(onResultOK, null, 'iosrtcPlugin', 'new_RTCPeerConnection', [
+			this.pcId,
+			this.pcConfig,
+			pcConstraints
+		]);
 	}
-});
-
-RTCPeerConnection.prototype.createOffer = function (options) {
-	// Detect callback usage to assist 5.0.1 to 5.0.2 migration
-	// TODO remove on 6.0.0
-	Errors.detectDeprecatedCallbaksUsage('RTCPeerConnection.prototype.createOffer', arguments);
 
-	var self = this;
-
-	if (isClosed.call(this)) {
-		return;
+	get localDescription() {
+		return this._localDescription;
 	}
+	get connectionState(): RTCPeerConnectionState {
+		if (this.isClosed()) {
+			return 'closed';
+		}
 
-	debug('createOffer() [options:%o]', options);
+		switch (this.iceConnectionState) {
+			case 'checking':
+				return 'connecting';
+			case 'completed':
+				return 'connected';
+		}
```

Comment:
`iceConnectionState` and `connectionState` are not exactly identical.  This converts the few different states based on the spec.

Reply:
ok